src/document/document.controller.ts

```diff
@@ -0,0 +1,75 @@
+import { Controller,
+  Post,
+  Req,
+  Res,
+  UseInterceptors,
+  UploadedFile,
+} from '@nestjs/common';
+import { ConfigService } from '../config/config.service';
+import { FileInterceptor } from '@nestjs/platform-express';
+import { Request } from 'express';
+import { CRMWebAPI } from '../_utils/crm-web-api';
```

Comment:
I think rather than importing this directly, we should use the internal OData module.

Reply:
Like import Odata instead of ConfigService?

Reply:
@godfreyyeung I guess so? It depends on whether you use ConfigService for something else in the module... my point is that instead of using CRMWebAPI directly, you use the OData module that I built (which wraps that).